library/core/class.oauth2.php

```diff
@@ -630,14 +630,15 @@ public function entryEndpoint($sender, $code, $state = '') {
             default:
 
                 // Save the access token and the profile to the session table, set expiry to 3 minutes.
+                $expiryTime = new \DateTimeImmutable('now + 5 minutes');
                 $stashID = $this->sessionModel->insert(
                     [
                         'Attributes' => [
                                 'AccessToken' => $response['access_token'] ,
                                 'RefreshToken' => $response['refresh_token'],
                                 'Profile' => $profile,
                             ],
-                        'DateExpires' => date(MYSQL_DATE_FORMAT, strtotime('3 minutes')),
+                        'DateExpires' => $expiryTime->format(MYSQL_DATE_FORMAT),
```

Comment:
```suggestion
                        'DateExpires' => $expiryTime,
```

Can you see if this works?

Reply:
I tried it. The Session Insert class takes a formatted datetime string, $expiryTime is an object.